### PR TITLE
Frame with embedded javascript

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
### What does this PR do?
Add an example demonstrates how to embed javascript using iframe.

framedjs

### What issues does this PR fix or reference?
https://github.com/codenvy/enterprise/issues/75

#### Changelog
Add option to create an example with framed javascript.

#### Release Notes
Added example with javascript embedded in iframe.

### CODE REVIEW

**Comment by skabashnyuk** [comment]
link to changes in cli?

**Comment by vitaliy-guliy** [comment]
Changes in CLI
https://github.com/eclipse/che/pull/5049

File: plugin-framedjs-archetype/pom.xml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    <parent>
+        <artifactId>eclipse-che-archetype-parent</artifactId>
+        <groupId>org.eclipse.che.archetype</groupId>
+        <version>5.10.0-SNAPSHOT</version>
+    </parent>
+    <groupId>org.eclipse.che.archetype</groupId>
+    <artifactId>plugin-framedjs-archetype</artifactId>
+    <packaging>maven-archetype</packaging>
+    <name>plugin-framedjs-archetype</name>
+    <description>Contains sample with menu entries that opening frame with java script UML example and communicating with java script inside frame</description>
```
Comment (ashumilova) [inline plugin-framedjs-archetype/pom.xml:25]: that opening frame = that open frame

File: plugin-framedjs-archetype/src/main/resources/archetype-resources/plugins/__rootArtifactId__/__rootArtifactId__-ide/src/main/java/__packageInPathFormat__/ide/view/UMLExampleViewImpl.ui.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+<ui:UiBinder xmlns:ui='urn:ui:com.google.gwt.uibinder'
+	xmlns:g='urn:import:com.google.gwt.user.client.ui'>
+	<ui:style>
+
+        .editorView {
+            width: 100%;
+            height: 100%;
+			overflow: auto;
+        }
+
```
Comment (ashumilova) [inline plugin-framedjs-archetype/src/main/resources/archetype-resources/plugins/__rootArtifactId__/__rootArtifactId__-ide/src/main/java/__packageInPathFormat__/ide/view/UMLExampleViewImpl.ui.xml:15]: formatting

File: plugin-framedjs-archetype/pom.xml (hunk 1 above)
Comment (ashumilova) [reply plugin-framedjs-archetype/pom.xml:25]: java script = javascript or JavaScript

File: plugin-framedjs-archetype/src/main/resources/archetype-resources/plugins/__rootArtifactId__/__rootArtifactId__-ide/src/main/java/__packageInPathFormat__/ide/view/client/jso/FrameOverlay.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+     * @param content new frame content
+     */
+    public native void setContent(String content) /*-{
+        var frame = this.@${package}.ide.view.client.jso.FrameOverlay::frame;
+        var doc = frame.contentDocument || frame.contentWindow.document;
+        doc.write(content);
+    }-*/;
+
+    /**
+     * Runs java script code inside frame.
```
Comment (ashumilova) [inline plugin-framedjs-archetype/src/main/resources/archetype-resources/plugins/__rootArtifactId__/__rootArtifactId__-ide/src/main/java/__packageInPathFormat__/ide/view/client/jso/FrameOverlay.java:106]: java script = javascript or JavaScript

**Review by ashumilova — APPROVED** [review]

**Comment by vitaliy-guliy** [comment]
Merging postponed due to the issue with panels (https://github.com/eclipse/che/issues/4929).
Also I have to create CQ on javascript libraries used in examples. Will be done soon.

**Comment by vitaliy-guliy** [comment]
SQs for javascript libraries:
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14372 - Backbone 1.3.3
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14374 - jQuery 2.2.4
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14375 - Lodash 3.10.1
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14376 - Joint JS 1.1.0

**Comment by benoitf** [comment]
@vitaliy-guliy CQs seem ok ?

File: plugin-framedjs-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        xmlns="http://maven.apache.org/plugins/maven-archetype-plugin/archetype-descriptor/1.0.0"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <fileSets>
+        <fileSet filtered="false" encoding="UTF-8">
+            <directory></directory>
+            <includes>
+                <include>build.include</include>
+                <include>build.sh</include>
+                <include>run.sh</include>
+                <include>stop.sh</include>
```
Comment (mkuznyetsov) [inline plugin-framedjs-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml:25]: build.include and .*sh scripts no longer exist in resources, these lines should be deleted

File: plugin-framedjs-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+                                    <includes>
+                                        <include>**/*.svg</include>
+                                        <include>**/*.xml</include>
+                                        <!--<include>**/*.js</include>-->
+                                    </includes>
+                                </fileSet>
+                                <fileSet filtered="false" encoding="UTF-8">
+                                    <directory>src/main/resources</directory>
+                                    <includes>
+                                        <!--<include>**/*.svg</include>-->
```
Comment (mkuznyetsov) [inline plugin-framedjs-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml:56]: is this commented code is needed?

File: plugin-framedjs-archetype/pom.xml [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    Contributors:
+      Codenvy, S.A. - initial API and implementation
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>eclipse-che-archetype-parent</artifactId>
+        <groupId>org.eclipse.che.archetype</groupId>
+        <version>5.10.0-SNAPSHOT</version>
```
Comment (mkuznyetsov) [inline plugin-framedjs-archetype/pom.xml:19]: use latest che master version here

**Review by mkuznyetsov — APPROVED** [review]
other ok

File: plugin-framedjs-archetype/pom.xml (hunk 1 above)
Comment (vitaliy-guliy) [reply plugin-framedjs-archetype/pom.xml:25]: Done

File: plugin-framedjs-archetype/src/main/resources/archetype-resources/plugins/__rootArtifactId__/__rootArtifactId__-ide/src/main/java/__packageInPathFormat__/ide/view/UMLExampleViewImpl.ui.xml (hunk 2 above)
Comment (vitaliy-guliy) [reply plugin-framedjs-archetype/src/main/resources/archetype-resources/plugins/__rootArtifactId__/__rootArtifactId__-ide/src/main/java/__packageInPathFormat__/ide/view/UMLExampleViewImpl.ui.xml:15]: Done

File: plugin-framedjs-archetype/src/main/resources/archetype-resources/plugins/__rootArtifactId__/__rootArtifactId__-ide/src/main/java/__packageInPathFormat__/ide/view/client/jso/FrameOverlay.java (hunk 3 above)
Comment (vitaliy-guliy) [reply plugin-framedjs-archetype/src/main/resources/archetype-resources/plugins/__rootArtifactId__/__rootArtifactId__-ide/src/main/java/__packageInPathFormat__/ide/view/client/jso/FrameOverlay.java:106]: Done

**Comment by skabashnyuk** [comment]
would not fix

**Pull request closed without merging**